src/khepri_machine.erl

```diff
@@ -1194,6 +1287,89 @@ apply(
                      State
              end,
     Ret = {State1, Reply},
+    bump_applied_command_count(Ret, Meta);
+apply(
+  Meta,
+  #acquire_lock{lock = #khepri_lock{lock_id = LockId,
+                                    group_id = GroupId,
+                                    release_on_exit = ReleaseOnExit},
+                acquirer = Acquirer},
+  #?MODULE{locks = Locks} = State) ->
+    Ret = case maps:get(LockId, Locks, {GroupId, []}) of
+              {GroupId, Acquirers} ->
+                  Acquirer1 = {Acquirer, ReleaseOnExit},
+                  LockEntry = {GroupId, [Acquirer1 | Acquirers]},
+                  Locks1 = Locks#{LockId => LockEntry},
+                  State1 = State#?MODULE{locks = Locks1},
+                  Effects = case ReleaseOnExit of
+                                true ->
+                                    [{monitor, process, Acquirer}];
```

Comment:
Like with local locks, we should try to limit to one monitor per process if possible.